google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/UploadConversionEnhancement.java

```diff
@@ -0,0 +1,282 @@
+      String restatementCurrencyCode)
+      throws NoSuchAlgorithmException, UnsupportedEncodingException {
+    // [START create_adjustment]
+    // Creates a builder for constructing the enhancement adjustment.
+    ConversionAdjustment.Builder enhancementBuilder =
+        ConversionAdjustment.newBuilder()
+            .setConversionAction(ResourceNames.conversionAction(customerId, conversionActionId))
+            .setAdjustmentType(ConversionAdjustmentType.ENHANCEMENT)
+            // Enhancements MUST use order ID instead of GCLID date/time pair.
+            .setOrderId(orderId);
+
+    // Creates a SHA256 message digest for hashing user identifiers in a privacy-safe way, as
+    // described at https://support.google.com/google-ads/answer/9888656.
+    MessageDigest sha256Digest = MessageDigest.getInstance("SHA-256");
+
+    // Adds user identifiers, hashing where required.
+
+    // Creates a user identifier using sample values for the user address.
+    UserIdentifier addressIdentifier =
+        UserIdentifier.newBuilder()
+            // User identifiers for conversion enhancements MUST use first party data.
+            .setUserIdentifierSource(UserIdentifierSource.FIRST_PARTY)
+            .setAddressInfo(
+                OfflineUserAddressInfo.newBuilder()
+                    .setHashedFirstName(normalizeAndHash(sha256Digest, "Joanna"))
+                    .setHashedLastName(normalizeAndHash(sha256Digest, "Smith"))
+                    .setHashedStreetAddress(
+                        normalizeAndHash(sha256Digest, "1600 Amphitheatre Pkwy"))
+                    .setCity("Mountain View")
+                    .setState("CA")
+                    .setPostalCode("94043")
+                    .setCountryCode("US"))
+            .build();
+
+    // Creates a user identifier using the hashed email address.
+    UserIdentifier emailIdentifier =
+        UserIdentifier.newBuilder()
+            .setUserIdentifierSource(UserIdentifierSource.FIRST_PARTY)
+            // Uses the normalize and hash method specifically for email addresses.
+            .setHashedEmail(normalizeAndHashEmailAddress(sha256Digest, "joannasmith@example.com"))
+            .build();
+
+    // Adds the user identifiers to the enhancement adjustment.
+    enhancementBuilder.addUserIdentifiers(addressIdentifier).addUserIdentifiers(emailIdentifier);
+
+    // Sets optional fields where a value was provided.
+
+    if (userAgent != null) {
+      // Sets the user agent. This should match the user agent of the request that sent the original
+      // conversion so the conversion and its enhancement are either both attributed as same-device
+      // or both attributed as cross-device.
+      enhancementBuilder.setUserAgent(userAgent);
+    }
+
+    if (restatementValue != null) {
+      // Creates a builder to construct the restated conversion value.
+      RestatementValue.Builder valueBuilder = enhancementBuilder.getRestatementValueBuilder();
```

Comment:
It doesn't look like valueBuilder builder is ever actually built. Is this correct? Does it work?